### ISSUE TITLE
Error parsing external relative ref that contains a local ref

### ISSUE THREAD

**Issue by kepler** [koxudaxi/datamodel-code-generator#714]
**Describe the bug**
Parsing fails with a `KeyError` when processing an external ref that contains a local ref.

```
/datamodel_code_generator/parser/jsonschema.py", line 54, in get_model_by_path
    return get_model_by_path(schema[keys[0]], keys[1:])
KeyError: 'model_a'
```

**To Reproduce**

With two schemas in separate paths, which are to be processed separately but can reference each other:
File `model_a/types.openapi.yaml`:
```yaml

 ---
openapi: 3.0.3
info:
  title: Model A definitions
  version: "1.0"

paths: {}


components:
  schemas:

    model_a.input:
      type: object
      properties:
        name:
          type: string

    model_a.output:
      type: object
      properties:
        output:
          type: string
        input:
          $ref: "#/components/schemas/model_a.input"

```

Running the following works as expected:
```
$ datamodel-codegen --input model_a --output tmp/
```

Now consider a second file, `model_b/module.openapi.yaml`, that references `model_a.input`:
```yaml

---
openapi: 3.0.3
info:
  title: Model B definitions
  version: "1.0"

paths: {}


components:
  schemas:
    modules.quality_evaluation.QualityEvaluationRequest:
      type: object
      properties:
        input:
          $ref: "../model_a/types.openapi.yaml#/components/schemas/model_a.input"

```

Running the following also works as expected:
```
$ datamodel-codegen --input model_b --output tmp/
```

But now, if instead we ref to `model_a.output`, which contains a ref itself:
```yaml

---
openapi: 3.0.3
info:
  title: Model B definitions
  version: "1.0"

paths: {}


components:
  schemas:
    modules.quality_evaluation.QualityEvaluationRequest:
      type: object
      properties:
        input:
          $ref: "../model_a/types.openapi.yaml#/components/schemas/model_a.output"

```

Then running:
```
$ datamodel-codegen --input model_b --output tmp/
```
breaks the following:
```
Traceback (most recent call last):
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/__main__.py", line 495, in main
    generate(
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/__init__.py", line 353, in generate
    results = parser.parse()
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/base.py", line 426, in parse
    self.parse_raw()
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/openapi.py", line 385, in parse_raw
    self.parse_raw_obj(
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1122, in parse_raw_obj
    self.parse_obj(name, JsonSchemaObject.parse_obj(raw), path)
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1142, in parse_obj
    self.parse_ref(obj, path)
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1084, in parse_ref
    self.parse_ref(value, path)
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1064, in parse_ref
    self.resolve_ref(obj.ref)
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1053, in resolve_ref
    self._parse_file(
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1266, in _parse_file
    models = get_model_by_path(raw, object_paths)
  File "/VENV/lib/python3.9/site-packages/datamodel_code_generator/parser/jsonschema.py", line 54, in get_model_by_path
    return get_model_by_path(schema[keys[0]], keys[1:])
KeyError: 'model_a'
```

**Expected behavior**
The local ref of the external ref is successfully parsed.

**Version:**
 - OS: Ubuntu 20.10
 - Python version: 3.9.8
 - datamodel-code-generator version: 0.11.15

**Additional context**
Please let me know if this is expected behavior (not to fully parse the whole schema file of external references).

**Comment by Niraj-Kamdar** [comment]
I am facing similar issue

**Comment by Niraj-Kamdar** [comment]
I am facing similar issue

**Comment by koxudaxi** [comment]
@kepler @Niraj-Kamdar 
I checked the problem.
In the current master branch, the problem is fixed.
I added a unittest for your case.
https://github.com/koxudaxi/datamodel-code-generator/pull/964

**Comment by dee-me-tree-or-love** [comment]
Hello @koxudaxi, thanks for looking into it! I have updated the version to `0.17.0`, but a problem seems to persist in my case. Would you have the time to take a look along? 

---

**Context info:**

`datamodel-code-generator` was installed with `poetry` for the project. 

```bash
$ poetry --version
$ poetry shell
...
$ datamodel-codegen --version
0.17.0
```

**Error example:**

One base schema:

```yaml
# ./common/cat.yaml
CatInfo:
  type: object
  required:
    - cat_id
  properties:
    cat_id:
      type: string
      description: ID of this cat
    details:
      $ref: "#/CatDetails"

CatDetails:
  type: object
  required:
    - name
    - birthYear
  properties:
    name:
      type: string
      description: Name of this cat
    birthYear:
      type: number
      description: Year of this cat's birth
```

Aggregation schema:

```yaml
# ./public/entities.yaml
openapi: 3.0.3
info:
  title: "Entity Schemas"
paths: {}
components:
  schemas:
    CatInfo:
      $ref: "../common/cat.yaml#/CatInfo"
```

Next, I run the `$ datamodel-codegen --input ./public/entities.yaml --output entities.py`, but I get the following error output:

```plaintext
Traceback (most recent call last):
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/__main__.py", line 671, in main
    generate(
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/__init__.py", line 413, in generate
    results = parser.parse()
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/base.py", line 952, in parse
    self.parse_raw()
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/openapi.py", line 440, in parse_raw
    self.parse_raw_obj(
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1395, in parse_raw_obj
    self.parse_obj(name, JsonSchemaObject.parse_obj(raw), path)
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1421, in parse_obj
    self.parse_ref(obj, path)
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1333, in parse_ref
    self.resolve_ref(obj.ref)
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1322, in resolve_ref
    self._parse_file(
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 1558, in _parse_file
    models = get_model_by_path(raw, object_paths)
  File "<...>/.venv/lib/python3.8/site-packages/datamodel_code_generator/parser/jsonschema.py", line 65, in get_model_by_path
    model = get_model_by_path(schema[keys[0]], keys[1:])  # type: ignore
KeyError: 'common'
```

**Modifications that make it successful:**

> One way to make it succeed - is to _place both files in the same directory, then all the referencing works as expected_.  
> If keeping the directory structure, then removing the internal reference in `./common/cat.yaml` seems to make it work.

One base schema _without internal refs_:

```yaml
# ./common/cat.yaml
CatInfo:
  type: object
  required:
    - cat_id
  properties:
    cat_id:
      type: string
      description: ID of this cat
```

Aggregation schema _is the same_:

```yaml
# ./public/entities.yaml
openapi: 3.0.3
info:
  title: "Entity Schemas"
paths: {}
components:
  schemas:
    CatInfo:
      $ref: "../common/cat.yaml#/CatInfo"
```

Next, I run the `$ datamodel-codegen --input ./public/entities.yaml --output entities.py`, and get a successful result:

```python
# generated by datamodel-codegen:
#   filename:  entities.yaml
#   timestamp: <...>

from __future__ import annotations

from pydantic import BaseModel, Field


class CatInfo(BaseModel):
    cat_id: str = Field(..., description='ID of this cat')
```

---

It might be that I am doing something wrong here, thanks a lot for your time in any case!  
If I find a solution in the meantime, I will post the news here.  
:v:

**Comment by koxudaxi** [comment]
@dee-me-tree-or-love 

I have released the fix version `0.17.1`
Thank you very much!!

**Comment by dee-me-tree-or-love** [comment]
> @dee-me-tree-or-love
> 
> I have released the fix version `0.17.1` Thank you very much!!

Great! Thanks a lot for the quick fix @koxudaxi!! Looking forward to using it 👍

**Comment by kepler** [comment]
Thank you, @koxudaxi!
